### PR TITLE
Multi parameter module

### PR DESCRIPTION
**Pull request by VincentRouvreau**
Comes on top of #976 
Maybe the doc to be reviewed or some part could be private.

### CODE REVIEW

File: src/Multi_persistence/include/gudhi/Line.h [hunk 1]
```diff
@@ -95,7 +98,7 @@ template <typename T> class Line {
 };
 template <typename T> inline bool Line<T>::check_direction() const {
   bool is_trivial=true;
-  for (const auto& stuff : basepoint_){
+  for (const auto& stuff : basepOne_critical_filtrationoint_){
```
Comment (hschreiber) [inline src/Multi_persistence/include/gudhi/Line.h:101]: A wrong paste?

File: src/Multi_persistence/include/gudhi/Line.h [hunk 2]
```diff
@@ -113,7 +116,7 @@ Line<T>::Line(const point_type &x, const point_type &v)
     : basepoint_(x), direction_(v) {check_direction();}
 
 
-template <typename T>
+template <typename T>One_critical_filtration
```
Comment (hschreiber) [inline src/Multi_persistence/include/gudhi/Line.h:119]: `One_critical_filtration` was probably wrongly paste here too, no?

**Comment by hschreiber** [comment]
I would put `Box.h` and `Line.h` in a subfolder such that we have to write `#include <gudhi/Multi_persistence/Box.h>` instead of `#include <gudhi/Box.h>`. Except if we think that it is an object general enough to be put in common?

File: src/Simplex_tree/example/simplex_tree_multi.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	static const bool contiguous_vertices = false;
+	static const bool link_nodes_by_label = true;
+	static const bool stable_simplex_handles = false;
+	static const bool is_multi_parameter = true;
+};
+
+using ST = Gudhi::Simplex_tree<ST_MULTI>;
+
+
+int main() {
```
Comment (hschreiber) [inline src/Simplex_tree/example/simplex_tree_multi.cpp:35]: Shouldn't the example be added in `src/common/doc/examples.h` or is it only useful for examples appearing in the introduction?

File: src/Multi_persistence/include/gudhi/Line.h (hunk 1 above)
Comment (VincentRouvreau) [reply src/Multi_persistence/include/gudhi/Line.h]: Thanks ! I have noticed that my IDE (Visual Studio Code) was copying things randomly (maybe when scrolling files, I may click). I changed my IDE. To be continued...
Done on [a42608e](https://github.com/GUDHI/gudhi-devel/pull/1123/commits/a42608ece812a62c49dbb145fbfef82cd30d9129)

File: src/Multi_persistence/include/gudhi/Line.h (hunk 2 above)
Comment (VincentRouvreau) [reply src/Multi_persistence/include/gudhi/Line.h]: Done on [a42608e](https://github.com/GUDHI/gudhi-devel/pull/1123/commits/a42608ece812a62c49dbb145fbfef82cd30d9129)

**Comment by VincentRouvreau** [comment]
> I would put `Box.h` and `Line.h` in a subfolder such that we have to write `#include <gudhi/Multi_persistence/Box.h>` instead of `#include <gudhi/Box.h>`. Except if we think that it is an object general enough to be put in common?

That's a good idea as the name is quite generic. I moved them on [893fbf6](https://github.com/GUDHI/gudhi-devel/pull/1123/commits/893fbf66ac3168df482b9d5d73e4f02acd179a09)

File: src/Simplex_tree/example/simplex_tree_multi.cpp (hunk 3 above)
Comment (VincentRouvreau) [reply src/Simplex_tree/example/simplex_tree_multi.cpp:35]: I added the example in the documentation on [ffaf004](https://github.com/GUDHI/gudhi-devel/pull/1123/commits/ffaf00448f552cb50eacf94422c88b80a9e595f8)

**Comment by VincentRouvreau** [comment]
Will be merged in [DavidLapous:Multiparameter-Simplextree-C++-part](https://github.com/DavidLapous/gudhi-devel-multi/tree/Multiparameter-Simplextree-C%2B%2B-part)

**Pull request closed without merging**